fml/message_loop_unittests.cc

```diff
@@ -315,3 +316,32 @@ TEST(MessageLoop, CanCreateConcurrentMessageLoop) {
   latch.Wait();
   ASSERT_GE(thread_ids.size(), 1u);
 }
+
+TEST(MessageLoop, TIME_SENSITIVE(WakeUpTimersAreSingletons)) {
+  auto loop_impl = fml::MessageLoopImpl::Create();
+
+  const auto t1 = fml::TimeDelta::FromMilliseconds(10);
+  const auto t2 = fml::TimeDelta::FromMilliseconds(20);
+
+  const auto begin = fml::TimePoint::Now();
+
+  // Register a task scheduled for 10ms in the future. This schedules a
+  // WakeUp call on the MessageLoopImpl with that fml::TimePoint
+  loop_impl->PostTask(
+      [&]() {
+        auto delta = fml::TimePoint::Now() - begin;
+        auto ms = delta.ToMillisecondsF();
+        ASSERT_GE(ms, 18);
+        ASSERT_LE(ms, 22);
+
+        loop_impl->Terminate();
+      },
+      fml::TimePoint::Now() + t1);
+
+  // Call WakeUp manually to change the WakeUp time to the future. If the
+  // timer is correctly set up to be rearmed instead of a new timer scheduled,
+  // the above task will be executed at t2 instead of t1 now.
+  loop_impl->WakeUp(fml::TimePoint::Now() + t2);
```

Comment:
Nit: `begin + t2`

Reply:
+1.  Not a nit either.  If the current thread is descheduled during the PostTask call  Now() and begin could be quite different and lead to a flake.

Same reasoning applies to the Now() + t1 above (although might be less likely to occur)